### ISSUE TITLE
Generators cannot be saved directly

### ISSUE THREAD

**Issue by lesrose** [SYSTRAN/faster-whisper#892]
Can directly return a list or other serializable types? The returned generator cannot be placed into a queue, and converting it to a list is time-consuming.

**Comment by trungkienbkhn** [comment]
No, you need to wait to transcribe each segment (30s) via this [logic](https://github.com/SYSTRAN/faster-whisper/blob/master/faster_whisper/transcribe.py#L789)

**Comment by lesrose** [comment]
> No, you need to wait to transcribe each segment (30s) via this [logic](https://github.com/SYSTRAN/faster-whisper/blob/master/faster_whisper/transcribe.py#L789)

It takes about 100 seconds to iterate through this generator, which is time-consuming. Is there a better way to do this?

**Comment by trungkienbkhn** [comment]
This is the required calculation time of the model. For acceleration, you can use smaller models (tiny, small, ...) or refer to use [multi threading](https://github.com/SYSTRAN/faster-whisper/issues/100#issuecomment-1492141352)

**Comment by lesrose** [comment]
> This is the required calculation time of the model. For acceleration, you can use smaller models (tiny, small, ...) or refer to use [multi threading](https://github.com/SYSTRAN/faster-whisper/issues/100#issuecomment-1492141352)这是模型所需的计算时间。对于加速，您可以使用较小的模型（微型、小型……）或参考使用多线程
Is it necessary to use a GPU when iterating through the generator?

**Comment by lesrose** [comment]
> This is the required calculation time of the model. For acceleration, you can use smaller models (tiny, small, ...) or refer to use [multi threading](https://github.com/SYSTRAN/faster-whisper/issues/100#issuecomment-1492141352)

Is it necessary to use a GPU when iterating through the generator?
